tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerInvoker.cs

```diff
@@ -224,6 +310,22 @@ public static void EndMethod_EndMarker(ref MethodDebuggerState state)
                 return;
             }
 
+            if (state.Condition?.ShouldEvaluate == true)
+            {
+                state.Condition.SetInvocationTarget("this", state.MethodMetadataInfo.DeclaringType, state.InvocationTarget);
+                var result = state.Condition.Evaluate();
+                if (!result)
+                {
+                    state.IsActive = false;
+                    return;
+                }
+
+                EndMethodStartMarkerCapture(
+                    state.Condition?.ScopeMembers.SingleOrDefault(m => m.ElementType == ScopeMember.MemberType.Return).Value,
+                    (Exception)state.Condition?.ScopeMembers.SingleOrDefault(m => m.ElementType == ScopeMember.MemberType.Exception).Value,
```

Comment:
Consider creating a record called `EvaluationScope` that would store all of the `ScopeMember` and adding specialized methods to get these special members (such as the return value, the exception, the invocation target, etc.).